src/main/kotlin/nl/tudelft/ewi/se/ciselab/testgenie/actions/GenerateTestsUtils.kt

```diff
@@ -0,0 +1,163 @@
+package nl.tudelft.ewi.se.ciselab.testgenie.actions
+
+import com.intellij.openapi.editor.Caret
+import com.intellij.psi.PsiAnonymousClass
+import com.intellij.psi.PsiClass
+import com.intellij.psi.PsiElement
+import com.intellij.psi.PsiFile
+import com.intellij.psi.PsiMethod
+import com.intellij.psi.util.PsiTreeUtil
+import com.intellij.refactoring.suggested.endOffset
+import com.intellij.refactoring.suggested.startOffset
+
+/**
+ * This class contains some useful methods related to GenerateTests actions.
+ */
+object GenerateTestsUtils {
+
+    /**
+     * Gets the class on which the user has clicked (the click has to be inside the contents of the class).
+     * NB! This has to be a concrete class, so enums, abstract classes and interfaces do not count.
+     *
+     * @param psiFile the current PSI file (where the user makes a click)
+     * @param caret the current (primary) caret that did the click
+     * @return PsiClass element if it has been found, null otherwise
+     */
+    fun getSurroundingClass(psiFile: PsiFile, caret: Caret): PsiClass? {
+        // Get the classes of the PSI file
+        val classElements = PsiTreeUtil.findChildrenOfAnyType(psiFile, PsiClass::class.java)
+
+        // Get the surrounding PSI class (i.e. the cursor has to be within that class)
+        var surroundingClass: PsiClass? = null
```

Comment:
can't you stop the loop when you found the first hit?

Reply:
No, otherwise you will ignore the nested things, like e.g nested static classes

Reply:
aaaaaaaaaaa, got it 👍